### DISCUSSION TITLE
Exposing a single copy of react/react-dom

### DISCUSSION THREAD

**Discussion by logan2z** [module-federation/module-federation-examples#954] [Q&A]
I'm trying to expose a single copy of react/react-dom that can be shared by all of the MFEs in my application. I looked at the complete-react-case example but I'm curious about one thing - does every MFE need to use imports of the following form in order to use the exposed version of those libraries?

import 'lib_app/react'

If so, what about third-party libraries that are imported that themselves import react? There's no way to change those import statements to reference lib_app. Is there another way to expose/share a common instance of react without modifying the import statements? I tried to alias 'react' to 'lib_app/react' but the import 'react' statements failed to compile since 'react' couldn't be resolved.

I feel like I'm missing something obvious.

**Comment by Gentlemancj** [comment]
i have same question

**Comment by Gentlemancj** [comment]
dou you have any solutions?
